pallets/base-nft/src/lib.rs

```diff
@@ -279,22 +285,29 @@ impl<T: Config> Pallet<T> {
 
 			let token_info = TokenInfo {
 				metadata,
-				owner: owner.clone(),
+				owners: owners.into_iter().cloned().collect(),
 				data,
 			};
+			let owners_token = token_info.owners.clone();
+
 			Tokens::<T>::insert(class_id, token_id, token_info);
 			#[cfg(not(feature = "disable-tokens-by-owner"))]
-			TokensByOwner::<T>::insert(owner, (class_id, token_id), ());
+			TokensByOwner::<T>::insert(owners_token, (class_id, token_id), ());
```

Comment:
Now we're storing it by all the owners, and we're storing the new field `fraction_owned` indicating the percentage of ownership. We've talked about moving forward with this approach for this pr specifically, but I still have questions.

We had mentioned an approach where each fractional NFT created from a parent NFT is just a separate NFT with a fraction assigned. To support a flow like that I would imagine one of the functions here would create multiple NFTs based on the percentage.  I would be curious to hear the team's thoughts on where/how this should be done.

Reply:
One more question is where does the interaction with IPFS happen? Thinking even further out -  should the file stored in IPFS be split up proportionally by owner and percentage owned?

Reply:
My thoughts on this would be:

- Alice upload a file to IPFS (our website)
- IPFS response with file location (CID)
- User click mint this Nft, polkadot wallet is asking him to sign the transaction
- Substrate mint's an Nft and set
pseudo code:
`TokensByOwner::insert(Alice, (class_id, token_id), TokenByOwnerData {fraction_owned: 100%})`

- Alice decides to transfer 20% of her token to Bob
pseudo code:
`TokensByOwner::set(Alice, (class_id, token_id), TokenByOwnerData {fraction_owned: 80%}))`
`TokensByOwner::set(Bob, (class_id, token_id), TokenByOwnerData {fraction_owned: 20%}))`


If above make sense, then we do not need `fraction_owned` at `TokenData` itself. 
Token data would be set only once (same IPFS CID forever), but information of its owners will be set on blockchain at `TokensByOwner` => `TokenByOwnerData` map.

Reply:
It makes sense. It seems less duplicative to manage the ownership in the storage map you pointed out.

Reply:
It might make sense to store structs if we're going to store more than one field. In this case, I think it's probably safe to assume that `TokenByOwnerData` would grow to hold more fields.